File: web/src/types/drone.ts
export interface DroneState {
  droneId: string
  lng: number
  lat: number
  alt: number
  heading: number
  pitch: number
  roll: number
  speed: number
  battery: number
  timestamp: number
  status?: string
  waypointIndex?: number
}

export type DroneFleet = Record<string, DroneState>

export interface Waypoint {
  lng: number
  lat: number
  alt: number
}

export interface DroneConfig {
  id: string
  name: string
  color: string
  speed: number
  route: Waypoint[]
}

export interface Mission {
  name: string
  drones: DroneConfig[]
}

File: api/model/drone.go
package model

type DroneState struct {
	DroneID       string  `json:"droneId"`
	Lng           float64 `json:"lng"`
	Lat           float64 `json:"lat"`
	Alt           float64 `json:"alt"`
	Heading       float64 `json:"heading"`
	Pitch         float64 `json:"pitch"`
	Roll          float64 `json:"roll"`
	Speed         float64 `json:"speed"`
	Battery       float64 `json:"battery"`
	Timestamp     int64   `json:"timestamp"`
	Status        string  `json:"status"`
	WaypointIndex int     `json:"waypointIndex"`
}

type Waypoint struct {
	Lng float64
	Lat float64
	Alt float64
}
